keras_cv/models/segmentation/deeplabv3plus.py

```diff
@@ -0,0 +1,354 @@
+
+        output = spatial_pyramid_pooling(high_level)
+        output = tf.keras.layers.UpSampling2D(
+            size=(4, 4),
+            interpolation="bilinear",
+        )(output)
+
+        low_level = layers.Conv2D(
+            filters=48,
+            kernel_size=1,
+            name="project_conv_bn_act",
+            use_bias=False,
+        )(low_level)
+        low_level = layers.BatchNormalization()(low_level)
+        low_level = layers.Activation("relu")(low_level)
+
+
+        output = layers.Concatenate()([output, low_level])
+
+        if segmentation_head is None:
+            segmentation_head = SegmentationHead(
+                classes=classes,
+                name="segmentation_head",
+                conv_type="conv2d",
+                output_scale_factor=4,
+                filters=256,
+                convs=2,
+                dropout=0.2,
+                kernel_size=3,
+            )
+
+        output = segmentation_head(output)
+
+        super().__init__(
+            inputs={
+                "inputs": inputs,
+            },
+            outputs={
+                "output": output,
+            },
+            **kwargs,
+        )
+
+        # All references to `self` below this line
+        self.classes = classes
+        self.backbone = backbone
+        self.spatial_pyramid_pooling = spatial_pyramid_pooling
+        self.segmentation_head = segmentation_head
+        self.segmentation_head_activation = segmentation_head_activation
+        self.layer_names = layer_names
+
+    def get_config(self):
+        return {
+            "classes": self.classes,
+            "backbone": self.backbone,
+            "spatial_pyramid_pooling": self.spatial_pyramid_pooling,
+            "segmentation_head": self.segmentation_head,
+            "segmentation_head_activation": self.segmentation_head_activation,
+            "layer_names": self.layer_names,
+        }
+
+    def compile(self, weight_decay=0.0001, **kwargs):
+        self.weight_decay = weight_decay
+        super().compile(**kwargs)
+
+    def train_step(self, data):
+        images, y_true, sample_weight = tf.keras.utils.unpack_x_y_sample_weight(data)
+        with tf.GradientTape() as tape:
+            y_pred = self(images, training=True)
+            total_loss = self.compute_loss(images, y_true, y_pred, sample_weight)
+            reg_losses = []
+            if self.weight_decay:
+                for var in self.trainable_variables:
+                    if "bn" not in var.name:
+                        reg_losses.append(self.weight_decay * tf.nn.l2_loss(var))
+                l2_loss = tf.math.add_n(reg_losses)
+                total_loss += l2_loss
+        self.optimizer.minimize(total_loss, self.trainable_variables, tape=tape)
+        return self.compute_metrics(images, y_true, y_pred, sample_weight=sample_weight)
+
+
+class SegmentationHead(layers.Layer):
```

Comment:
It does seem we have a need to at least unify these two segmentation heads, into models/segmentation/segmentation_head.py, as long as we don't expose this as public API yet.
WDYT?

Reply:
And add unit test for it.

Reply:
Sounds good! I've been hoping to combine it :)
Updating this and DLV3